effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -675,50 +688,106 @@ object Namer extends Phase[Parsed, NameResolved] {
    * We de-alias on-the-fly in Namer so that aliases can never show up again in the remaining compiler.
    * This way error messages might suffer; however it simplifies the compiler a lot.
    */
-  def resolve(tpe: source.ValueType)(using Context): ValueType = resolvingType(tpe) {
-    case source.ValueTypeRef(id, args) => Context.resolveType(id) match {
-      case constructor: TypeConstructor => ValueTypeApp(constructor, args.map(resolve))
+  def resolveValueType(tpe: source.ValueType, isParam: Boolean = false)(using Context): ValueType = resolvingType(tpe) {
+    case source.TypeRef(id, args) => Context.resolveType(id) match {
+      case constructor: TypeConstructor => ValueTypeApp(constructor, args.map(resolveValueType))
       case id: TypeVar =>
         if (args.nonEmpty) {
           Context.abort(pretty"Type variables cannot be applied, but receieved ${args.size} arguments.")
```

Comment:
```suggestion
          Context.abort(pretty"Type variables cannot be applied, but received ${args.size} arguments.")
```

Reply:
I've noticed this, but I didn't want to do a drive-by on unrelated code :^)

<img width="1002" alt="Screenshot 2025-05-06 at 13 53 57" src="https://github.com/user-attachments/assets/2361ec6e-7c49-41b1-b500-aca79441458d" />

I'll fix it :)